documentation/writing_tests.md

```diff
@@ -152,28 +176,41 @@ Mocha is used by our test scripts in our `package.json` files. We typically invo
 
-Then we have to point mocha to our test files. If you're **not** using `nyc`, you can point to the bundled test file (bundled with rollup, which we will see later), typically at `dist-test/index.node.js`. If you are using `nyc`, point mocha to the files built by the TypeScript compiler, normally at `dist-esm/test/*.test.js`.
+Then we have to point mocha to our test files. If you're **not** using `nyc`, you can point to the bundled test file (bundled with rollup, which we will see later), typically at `dist-test/index.node.js`. If you are using `nyc`, point mocha to the files built by the TypeScript compiler, which can be found using `find dist-esm/test -name '*.spec.js'` before calling mocha.
+
+To use Mocha from the `package.json`, our systems will expect to encounter two scripts, one called `unit-test` for tests that will be [executed during Pull Request validation](https://github.com/Azure/azure-sdk-for-js/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L226-L233), which won't ever reach to live resources, and another called `integration-test` for our [nightly and release builds](https://github.com/Azure/azure-sdk-for-js/blob/master/eng/pipelines/templates/jobs/archetype-sdk-integration.yml#L114), which will be expected to reach to live resources. We assume that the distinction of when to reach to what resources will be done within the tests (either from  [The Recorder](#the-recorder) or through [Using conditionals](#using-conditionals)). With this in mind, and limiting `nyc` to only run on the `integration-test` script (since `unit-test` will be used for debugging), we will end up with the following scripts:
+
+```json
+    "integration-test:node": "find dist-esm/test -name '*.spec.js' | xargs nyc mocha -r esm --require source-map-support/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 180000 --full-trace",
+    "unit-test:node": "mocha --require source-map-support/register --reporter ../../../common/tools/mocha-multi-reporter.js --timeout 180000 --full-trace dist-test/index.node.js",
+```
+
+Keep in mind that Mocha will be directly called from our `package.json` scripts only for our **NodeJS** tests. For our browser tests, we will be using [Karma](#karma).
 
 You can look at how Mocha's configuration is present in our [template project's package.json file](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/template/template/package.json).
 
+#### Code coverage with Mocha and nyc
+
+Our `integration-test` script will output code-coverage on our nightly and release builds. It's recommended to run this manually to confirm that code coverage is high. Code coverage should be above 80%, though code coverage should not change how we write code or tests.
```

Comment:
> Code coverage should be above 80%

Maybe we should focus on this.. 
Ideally, it should be 100% and we need to ensure that each and every branch is covered.

Reply:
In my experience when you go for 100% you end up writing useless tests that get the coverage but don't test anything useful. Remember coverage doesn't imply that you assert anything, just that code ran. :)

Reply:
In particular this being an integration test means that 100% isn't feasible unless you can provoke every error condition, etc...

Unit-tests, however, should have a higher goal IMO. We just need to be reasonable about not requiring coverage in generated code, accessors, etc...

Reply:
We don't require code coverage. I've expanded on this topic to reflect that 80% is considered a good measure, but it's up to the good judgement of developers. If anybody has a clear strategy that can be enforced, please let us know.